src/ClassyResource/_makeRequest.js

```diff
@@ -11,70 +11,109 @@ import request from 'request';
-      ? null
-      : undefined;
-
-    const requestParams = {
-      baseUrl: this.baseUrl,
-      uri: path,
-      method: method,
-      headers: headers,
-      rejectUnauthorized: false,
-      form: form,
-      encoding: encoding
-    };
-
-    if (method === 'GET') {
-      requestParams.qs = data;
-    } else {
-      requestParams.body = JSON.stringify(data);
+    if (data.noLog) {
+      headers['x-no-log'] = true;
+      delete data.noLog;
     }
 
-    if (forceQs) {
-      requestParams.qs = _.merge(requestParams.qs || {}, forceQs);
+    if (_.isObject(data.qs)) {
+      forceQs = data.qs;
+      delete data.qs;
     }
 
-    request(requestParams, (err, response, body) => {
-      if (err || !/^2/.test('' + response.statusCode)) {
-        let error;
-        if (err && err instanceof Error) {
-          error = err;
-        } else if (err && !(err instanceof Error)) {
-          const errorString = JSON.stringify(err);
-          error = new Error(errorString);
-        } else {
-          const errorString = body ? JSON.stringify(body) : 'Non-200 response';
-          error = new Error(errorString);
-          error.statusCode = response.statusCode;
-        }
+    const promise = new Promise((resolve, reject) => {
+      // the encoding param defaults to undefined, which means the response will be stringified
+      // we want to preserve the binary encoding for pdfs, so return null in this case
+      const encoding = _.includes(headers.Accept, 'pdf')
+        ? null
+        : undefined;
 
-        reject(error);
+      const requestParams = {
+        baseUrl: this.baseUrl,
+        uri: path,
+        method: method,
+        headers: headers,
+        rejectUnauthorized: false,
+        form: form,
+        encoding: encoding
+      };
+
+      if (method === 'GET') {
+        requestParams.qs = data;
       } else {
-        // if we're not returning a pdf file, parse body
-        if (!_.includes(response.headers['content-type'], 'pdf')) {
-          body = body ? JSON.parse(body) : {};
-        }
+        requestParams.body = JSON.stringify(data);
+      }
 
-        resolve(body);
+      if (forceQs) {
+        requestParams.qs = _.merge(requestParams.qs || {}, forceQs);
       }
+
+      request(requestParams, (err, response, body) => {
+        if (err || !/^2/.test('' + response.statusCode)) {
+          let error;
+
+          if (err && err instanceof Error) {
+            error = err;
+          } else if (err && !(err instanceof Error)) {
+            const errorString = JSON.stringify(err);
+
+            error = new Error(errorString);
+          } else {
+            const errorString = body ? JSON.stringify(body) : 'Non-200 response';
+
+            error = new Error(errorString);
+            error.statusCode = response ? response.statusCode : undefined;
+          }
+
+          /**
+           * Pass the error to _errorLogger if defined
+           */
+          if (this._errorLogger) {
```

Comment:
I wonder if we should be logging all errors or maybe exclude 403s and 404s?

Reply:
Hmm, not sure. I feel like that would maybe be up to the client (implementing errorLogger) to decide.

Reply:
good point good point